runner/utils/driver/istio/istio.go

```diff
@@ -66,7 +67,23 @@ func NewIstioDriver(auth *auth.Module, c *Config) (*Istio, error) {
 	// Start the keda external scaler
 	go kedaScaler.Start()
 
-	return &Istio{auth: auth, config: c, kube: kube, istio: istio, keda: kedaClient, kedaScaler: kedaScaler}, nil
+	waitservice := make(map[string]Deployment)
+	if err := WatchDeployments(func(eventType string, availableReplicas, readyReplicas int32, projectID, deploymentID string) {
+		waitservice = map[string]Deployment{
+			projectID: Deployment{
+				DeployemtID: map[string]Replicas{
+					deploymentID: Replicas{
+						AvailableReplicas: availableReplicas,
+						ReadyReplicas:     readyReplicas,
+					},
+				},
+			},
+		}
```

Comment:
You should have a switch case here on the variable eventType
case add,update:
    // add/update in map
case delete:
    // delete from map

& the logic for adding is wrong. you have to check if project & deployment exists